prody/ensemble/functions.py

```diff
@@ -362,18 +363,40 @@ def buildPDBEnsemble(refpdb, PDBs, title='Unknown', labels=None, seqid=94, cover
     :arg unmapped: A list of PDB IDs that cannot be included in the ensemble. This is an 
     output argument. 
     :type unmapped: list
+
+    :arg alignment_list: list of alignment duplets for mapping function
+        default is list of None
+    :type alignment_list: list
+
+    :arg alignments_list: list of alignments for mapping function
+        default is list of None
+    :type alignments_list: list
     """
 
     if not isinstance(refpdb, (Chain, Segment, Selection, AtomGroup)):
         raise TypeError('Refpdb must be a Chain, Segment, Selection, or AtomGroup.')
     
     if labels is not None:
         if len(labels) != len(PDBs):
-            raise TypeError('Labels and PDBs must have the same lengths.')
+            raise TypeError('labels and PDBs must have the same lengths.')
+
+    alignment_list = kwargs.get('alignment_list', None)
```

Comment:
This processing is not necessary if the user did not provide alignment_list. alignment_list can be wrapped into kwargs and given to mapping_func

Reply:
mapping_func is designed to be generic to all mapping functions including user-defined ones, so it should be as simple as possible. Ideally, only the four arguments are necessary, reference, target, seqid, and coverage. All the extra arguments that are needed for variants of mapping functions should go to kwargs to provide the flexibility and keep the simplicity at the same time.